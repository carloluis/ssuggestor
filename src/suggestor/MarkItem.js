import React from 'react';

const MarkItem = ({ item, search }) => {
	if (!search) {
		return <a>{ item }</a>;
	}

	let index = item.toLowerCase().indexOf(search.toLowerCase());
	if (index === -1) {
		return <a>{ item }</a>;	
	}

	let searchLength = search.length;

	return (
		<a>
			<span>
				{ item.substr(0, index) }
				<strong>{ item.substr(index, searchLength) }</strong>
				{ item.substr(index+searchLength, item.length) }
			</span>
		</a>
	);
};
MarkItem.propTypes = {
	item: React.PropTypes.string.isRequired,
	search: React.PropTypes.string.isRequired
};

export default MarkItem;