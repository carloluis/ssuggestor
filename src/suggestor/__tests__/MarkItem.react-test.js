import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import MarkItem from '../MarkItem';

describe('<MarkItem />', () => {
	let props;
	beforeEach(() => {
		props = { item: 'abcdefghijklmnopqrstuvwxyz', search: '' };
	});
	it('if no search => render plain item (<a>{item}</a>)', () => {
		const tree = renderer.create(<MarkItem {...props} />);
		expect(tree).toMatchSnapshot();
	});
	it('if search but no match => render plain item', () => {
		let _props = { ...props, search:'no-match' };
		const tree = renderer.create(<MarkItem {..._props} />);
		expect(tree).toMatchSnapshot();
	});
	it('if search, then render pattern inside <strong>', () => {
		let _props = { ...props, search:'x' };
		const tree = renderer.create(<MarkItem {..._props} />);
		expect(tree).toMatchSnapshot();
	});
});