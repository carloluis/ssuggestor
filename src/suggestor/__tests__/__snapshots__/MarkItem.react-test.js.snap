exports[`<MarkItem /> if no search => render plain item (<a>{item}</a>) 1`] = `
<a>
  abcdefghijklmnopqrstuvwxyz
</a>
`;

exports[`<MarkItem /> if search but no match => render plain item 1`] = `
<a>
  abcdefghijklmnopqrstuvwxyz
</a>
`;

exports[`<MarkItem /> if search, then render pattern inside <strong> 1`] = `
<a>
  <span>
    abcdefghijklmnopqrstuvw
    <strong>
      x
    </strong>
    yz
  </span>
</a>
`;
